src/main/java/io/nats/client/impl/NatsWatchSubscription.java

```diff
@@ -30,25 +32,29 @@ public NatsWatchSubscription(JetStream js) {
         this.js = js;
     }
 
-    protected void finishInit(NatsFeatureBase fb, String subscribeSubject, DeliverPolicy deliverPolicy, boolean headersOnly, WatchMessageHandler<T> handler)
+    protected void finishInit(NatsFeatureBase fb, String subscribeSubject, DeliverPolicy deliverPolicy, boolean headersOnly, long fromRevision, WatchMessageHandler<T> handler)
         throws IOException, JetStreamApiException
     {
-        if (deliverPolicy == DeliverPolicy.New
-            || fb._getLast(subscribeSubject) == null)
-        {
-            handler.sendEndOfData();
+        if (fromRevision > ULONG_UNSET) {
+            deliverPolicy = DeliverPolicy.ByStartSequence;
+        }
+        else {
+            fromRevision = ULONG_UNSET; // easier on the builder since we aren't starting at a fromRevision
+            if (deliverPolicy == DeliverPolicy.New || fb._getLast(subscribeSubject) == null) {
+                handler.sendEndOfData();
+            }
         }
 
         PushSubscribeOptions pso = PushSubscribeOptions.builder()
             .stream(fb.getStreamName())
             .ordered(true)
-            .configuration(
-                ConsumerConfiguration.builder()
-                    .ackPolicy(AckPolicy.None)
-                    .deliverPolicy(deliverPolicy)
-                    .headersOnly(headersOnly)
-                    .filterSubject(subscribeSubject)
-                    .build())
+            .configuration(ConsumerConfiguration.builder()
+                .ackPolicy(AckPolicy.None)
+                .deliverPolicy(deliverPolicy)
+                .startSequence(fromRevision)
```

Comment:
Is it ok to always be setting this option (even to 0 if not used)?

Reply:
Yes. I check the value before setting it and the builder knows how to deal with it.